#! /usr/bin/env python

PACKAGE='header_manipulation'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("msg_delay_milliseconds", double_t, 0, "Timespan the message will be delayed.", 0.0, 0.0, 10000.0)
gen.add("time_offset_milliseconds", double_t, 0, "Time offset added to the header timestamp of the msg.", 2000.0, -1000.0, 1000.0)
gen.add("publish_retry_rate", double_t, 0, "Max rate at which the publish msgs loop will run in all concurrent publishMsg Threads.", 100.0, 0.0, 1000.0)

exit(gen.generate(PACKAGE, "TimeModification", "TimeModification"))
